### PR TITLE
Added basic URI generation, minting, and testing

### PR DESCRIPTION
**Pull request by JavierMNieto**
Closes #10 #5 #15 #3. Latest deploy at [0xe24c11f3E38EB2467d6DA05606E1E698cAcea28C](https://rinkeby.etherscan.io/address/0xe24c11f3E38EB2467d6DA05606E1E698cAcea28C#code). Can be viewed on OpenSea [here](https://testnets.opensea.io/collection/illiniblockchain-v3).

### CODE REVIEW

File: src/IlliniBlockchainSP22Token.sol [hunk 1]
```diff
@@ -2,47 +2,113 @@
...
     function mint(
-        uint8 _id,
+        address _to,
+        uint256 _id,
         uint256 _amount,
+        TokenMetadata calldata _metadata,
         bytes memory _data
     ) public onlyOwner {
-        _mint(msg.sender, _id, _amount, _data);
+        tokenMetadata[_id] = _metadata;
```
Comment (zfaizal2) [inline src/IlliniBlockchainSP22Token.sol]: Can previous IDs be overwritten this way? Do we need some kind of tokenID validation?

**Review by zfaizal2 — APPROVED** [review]
Looks good, made a minor change to allow the test to pass (reference URI was missing a quote in the JSON) -- not sure if we need to check IDs for overwritten data.

File: src/IlliniBlockchainSP22Token.sol (hunk 1 above)
Comment (JavierMNieto) [reply src/IlliniBlockchainSP22Token.sol]: Yeah, good point. Maybe there should just be a different function for setting the metadata. I guess I was treating this more as an ERC 721 contract and forgot that you can still mint more of a specific token id. We can have a `setTokenURI` function that can only be called once for a token.

Comment (zfaizal2) [reply src/IlliniBlockchainSP22Token.sol]: perfect, that's exactly what i was thinking actually -- at first i thought it's cumbersome to set URI, then mint -- but actually, it makes more sense than feeding the URI data on each mint.

**Pull request merged**